vendor/github.com/ory-am/dockertest/consul.go

```diff
@@ -0,0 +1,99 @@
+package dockertest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+)
+
+var (
+	// ConsulDatacenter must be defined when starting a Consul datacenter; this
+	// value will be used for both the datacenter and the ACL datacenter
+	ConsulDatacenter = "test"
+
+	// ConsulACLDefaultPolicy defines the default policy to use with Consul ACLs
+	ConsulACLDefaultPolicy = "deny"
+
+	// ConsulACLMasterToken defines the master ACL token
+	ConsulACLMasterToken = "test"
+
+	// A function with no arguments that outputs a valid JSON string to be used
+	// as the value of the environment variable CONSUL_LOCAL_CONFIG.
+	ConsulLocalConfigGen = DefaultConsulLocalConfig
+)
+
+func DefaultConsulLocalConfig() (string, error) {
+	type d struct {
+		Datacenter       string `json:"datacenter,omitempty"`
+		ACLDatacenter    string `json:"acl_datacenter,omitempty"`
+		ACLDefaultPolicy string `json:"acl_default_policy,omitempty"`
+		ACLMasterToken   string `json:"acl_master_token,omitempty"`
+	}
+
+	vals := &d{
+		Datacenter:       ConsulDatacenter,
+		ACLDatacenter:    ConsulDatacenter,
+		ACLDefaultPolicy: ConsulACLDefaultPolicy,
+		ACLMasterToken:   ConsulACLMasterToken,
+	}
+
+	ret, err := json.Marshal(vals)
+	if err != nil {
+		return "", err
+	}
+
+	return string(ret), nil
+}
+
+// SetupConsulContainer sets up a real Consul instance for testing purposes,
+// using a Docker container. It returns the container ID and its IP address,
+// or makes the test fail on error.
+func SetupConsulContainer() (c ContainerID, ip string, port int, err error) {
+	port = RandomPort()
+	forward := fmt.Sprintf("%d:%d", port, 8500)
+	advertise := ip
+	if BindDockerToLocalhost != "" {
+		advertise = "127.0.0.1"
+		forward = advertise + ":" + forward
+	}
+	localConfig, err := ConsulLocalConfigGen()
+	if err != nil {
+		return "", "", 0, err
+	}
+	c, ip, err = SetupContainer(ConsulImageName, port, 10*time.Second, func() (string, error) {
+		return run(
+			"--name", GenerateContainerID(),
+			"-d",
+			"-p", forward,
+			"-e", fmt.Sprintf("CONSUL_LOCAL_CONFIG=%s", localConfig),
+			ConsulImageName,
+			"agent",
+			"-server",                // Run in server mode
```

Comment:
You can replace `-server` and `-bootstrap-expect=1` with simply `-dev`. This has the web ui and some other nice things and will always have the easiest to configure standalone dev server config.